collector/topic/src/main/scala/no/sysco/middleware/kafka/metadata/collector/topic/TopicCollector.scala

```diff
@@ -1,8 +1,29 @@
 package no.sysco.middleware.kafka.metadata.collector.topic
 
-import akka.actor.ActorSystem
+import akka.actor.{ActorRef, ActorSystem}
 import akka.stream.ActorMaterializer
+import com.typesafe.config.{Config, ConfigFactory}
+import no.sysco.middleware.kafka.metadata.collector.topic.internal._
 
-class TopicCollector(system: ActorSystem, materializer: ActorMaterializer) {
+import scala.concurrent.ExecutionContext
 
+object TopicCollector {
+  implicit val system: ActorSystem = ActorSystem("kafka-metadata-collector-topic")
+  new TopicCollector(system)
+}
+
+class TopicCollector(system: ActorSystem) {
+
+  implicit val executionContext: ExecutionContext = system.dispatcher
+  implicit val materializer: ActorMaterializer = ActorMaterializer()(system)
+
+  val config: Config = ConfigFactory.load()
+  val appConfig: TopicCollectorConfig = new TopicCollectorConfig(config)
+
+  val topicEventProducer: ActorRef = system.actorOf(TopicEventProducer.props())
+  val topicRepository: ActorRef = system.actorOf(TopicRepository.props(appConfig.Kafka.bootstrapServers))
+  val topicManager: ActorRef = system.actorOf(TopicManager.props(appConfig.Collector.pollFrequency, topicRepository, topicEventProducer))
+  val topicEventConsumer: ActorRef = system.actorOf(TopicEventConsumer.props(topicManager))
+
+  topicManager ! CollectTopics
```

Comment:
I think, this is not necessary, you can send init-message inside actor to itself on `PostConstruct` phase and after forward message futher.

Reply:
Good, I will review this as part of the refactor or Actor hierarchy,